### PR TITLE
Document tilesets

### PR DESCRIPTION
**Pull request by lmoureaux**
This PR aims at drastically improving the documentation of tilesets by tackling the first two chapters of #774:

* [x] Bootstraping the tutorial: this includes locating the data path, creating the new tileset and selecting it in the menu.
* [x] Adding a new unit. This sounds like a good intro to the mechanics in use: interaction with the ruleset, `spec` files, grids

### CODE REVIEW

File: docs/Modding/Tilesets/Tutorial/02_units.rst [hunk 1]
```diff
@@ -0,0 +1,107 @@
+Adding units
+************
+
+One of the most common changes done when developing custom rules (as described in the
+:doc:`/Modding/Rulesets/overview`) is to add new unit types to the game. It is important to note that
+Freeciv21 has a priori no knowledge of which unit types will exist, and much less of what they should look
```
Comment (jwrober) [inline docs/Modding/Tilesets/Tutorial/02_units.rst:6]: has no prior knowledge

File: docs/Modding/Tilesets/Tutorial/02_units.rst [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+* :download:`aliens.png </_static/images/tileset-tutorial/aliens.png>` --- this is the same image as above.
+* :download:`aliens.spec` --- we'll go through it shortly.
+
+The ``aliens.spec`` file tells Freeciv21 which sprites are in the image, and where in the grid. There is only
+one small change left to make the new sprites available in our tileset: we need to add the image to the
+tileset. To do this, open ``tutorial.tilespec`` and find the line with ``files =`` (around the half of the
+file). Insert the new ``spec`` file at the top of the list:
+
+.. code-block:: py
```
Comment (jwrober) [inline docs/Modding/Tilesets/Tutorial/02_units.rst:49]: wouldn't this be ini instead of py?

File: docs/Modding/Tilesets/Tutorial/02_units.rst [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+.. figure:: /_static/images/tileset-tutorial/aliens_game.png
+  :alt: A game screenshot with three alien units, two green and one red.
+  :align: center
+
+  Alien invasion.
+
+Let's now go through the newly added ``aliens.spec`` file. It can be opened with your text editor; go back to
+the :doc:`first page <../tutorial>` if you don't know what to use.
+
+You will notice that the file contains four sections, each opened by a names between square brackets:
```
Comment (jwrober) [inline docs/Modding/Tilesets/Tutorial/02_units.rst:74]: singular "name"

File: docs/Modding/Tilesets/Tutorial/02_units.rst [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+Freeciv21 runs.
+
+.. literalinclude:: aliens.spec
+  :language: ini
+  :lines: 11-
+
+The last section describes the layout of the sprites within the image. There is a single grid, called
+``main``, that starts one pixel from the top and one pixel from the left (``x_top_left`` and ``y_top_left``).
+This is because of the green outline around the file: it is not counted as being part of the grid. The next
+two lines specify the width (``dx``) and the height (``dy``) of the sprites in pixels. Then, we instructs
```
Comment (jwrober) [inline docs/Modding/Tilesets/Tutorial/02_units.rst:97]: singular "instruct"

File: docs/Modding/Tilesets/tutorial.rst [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  Tutorial/02_units.rst
+
+.. rubric:: A note on graphics software
+
+If you create tilesets, you will likely also want to edit graphics. We are often asked which painting program
+we use. We believe that the program that will work best for you depends a lot on your workflow, so we
+encourage you to try several of them and make your own opinion. If you don't know where to start, here are a
+few free (as in speech) recommendations:
+
+* `Gimp`_ is a generic image editor with a focus on image edition. It also has great drawing capabilities.
```
Comment (jwrober) [inline docs/Modding/Tilesets/tutorial.rst:40]: I think that is "image editing"

**Review by jwrober — CHANGES_REQUESTED** [review]
A few misspellings to fix.

File: client/helpdata.cpp [hunk 6]
```diff
@@ -476,8 +476,35 @@ static bool insert_generated_text(char *outbuf, size_t outlen,
...
+        data_dirs_info += _("(system)");
+      }
+      data_dirs_info += "\n";
+    }
+    data_dirs_info += "\n";
+    data_dirs_info += _("Note that for multiplayer games, the server may "
+                        "use different paths.");
+
+    cat_snprintf(outbuf, outlen, "%s", qUtf8Printable(data_dirs_info));
+
```
Comment (jwrober) [inline client/helpdata.cpp:507]: Isn't all this in another PR?

**Review by jwrober — CHANGES_REQUESTED** [review]

File: client/helpdata.cpp (hunk 6 above)
Comment (lmoureaux) [reply client/helpdata.cpp:507]: It was split from this one but will be needed to bootstrap the tutorial (instructing the user to put the files in the right location)

File: docs/Modding/Tilesets/Tutorial/02_units.rst (hunk 2 above)
Comment (lmoureaux) [reply docs/Modding/Tilesets/Tutorial/02_units.rst:49]: py looks better in this case... I agree it's weird

**Comment by lmoureaux** [comment]
Fixed typos and added chapter 1. The tutorial should now be self-consistent (if short) so I think this can be merged after one more review.

**Review by jwrober — APPROVED** [review]

**Comment by lmoureaux** [comment]
`clang-format` failure tackled in #784

**Pull request merged**